lib/multi.c

```diff
@@ -1110,6 +1110,16 @@ CURLMcode curl_multi_fdset(struct Curl_multi *multi,
   return CURLM_OK;
 }
 
+#ifdef USE_WINSOCK
+static bool check_socktype_stream(SOCKET s)
+{
+  int t;
+  int l = (int)sizeof(t);
+  return (!getsockopt(s, SOL_SOCKET, SO_TYPE, (char *)&t, &l) &&
+          t == SOCK_STREAM);
```

Comment:
maybe this function should also do the send(), instead of leaving that done in two places where this function is called?

Reply:
Ok done. I should note, I am still unsure how this is supposed to work with sockets other than TCP and UDP that are provided by the user, or if it's even appropriate to send 0 in those cases.

Reply:
I don't think we will get to know that until we merge this fix and someone actually runs into a side-effect in a distant future. At least this fix has the potential to remove a lot of crazy from using h3 on Windows right now!